### PR TITLE
{% cache %} compat follow-ups: docs, callback leak fix, regression tests

### PR DESCRIPTION
**Pull request by JuroOravec**
(Juro)

As I read through #1619, I explored and found:
- 1 bug
- few edge cases that are still problematic with the compatibility with Django's `{% cache %}` tag.

I captured the "proper" solution here https://github.com/django-components/django-components/issues/1650.

But since the proper design would be possibly only in v3, right now I only documented what to be aware of when using the `{% cache %}` tag with components.

---

(Opus 4.8)

## Summary

Follow-ups to #1619 (Django `{% cache %}` tag compatibility with components):

- **Docs.** New "Using Django's `{% cache %}` tag with components" section in [docs/concepts/advanced/component_caching.md](docs/concepts/advanced/component_caching.md). Covers the three real gotchas users can hit (same cached fragment embedded twice on a page → handlers attached more than once; JS/CSS variables silently lost after a deploy with a persistent fragment cache; `vary_on` hygiene), and documents the process-global patch of Django's built-in `{% cache %}` tag.

- **Fix.** Bounded callback-dict accumulation in `_assemble_cached_fragment`: the two synthetic callbacks added to `tree.on_component_intermediate_callbacks` / `tree.on_component_rendered_callbacks` per `{% cache %}` invocation are now popped via `try/finally`, so a template-level loop with many `{% cache %}` calls doesn't leave one dead-lambda pair per iteration in the (still-live) tree's callback dicts.

- **Tests.** Two regression tests:
  - `test_cache_embedded_twice_on_same_page_freezes_inner_render_id` — pins the documented "frozen `data-djc-id`" behavior so a future change is forced to discuss it.
  - `test_error_in_cache_body_does_not_poison_cache` — verifies that an error mid-cache-body doesn't write a partial value (the second render must run the component again, not return a poisoned hit).

- **Docs/comments.** Rewrote the [cache_tag.py](src/django_components/cache_tag.py) module docstring (the old text described an earlier, abandoned implementation approach), and annotated the `DjcCacheNode.render` body to mark which lines are copied verbatim from Django's `CacheNode.render` (Django 5.2) and which are djc-specific. Helps future maintainers re-diff against upstream when Django bumps.

The architectural fix that would dissolve both documented limitations entirely (cache `RenderObject` instead of the rendered string) is tracked in #1650 for v3.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/django_components/cache_tag.py [hunk 1]
```diff
@@ -41,43 +72,76 @@
...
+            component_tree_context=component_ctx.tree,
+            on_component_tree_rendered=lambda html: html,
+        )
+    finally:
+        component_ctx.tree.on_component_intermediate_callbacks.pop(render_id, None)
+        component_ctx.tree.on_component_rendered_callbacks.pop(render_id, None)
 
 
 class DjcCacheNode(CacheNode):
     def render(self, context: Context) -> str:
```
Comment (JuroOravec) [inline src/django_components/cache_tag.py:126]: Added commnets here to it's clearer which parts were copied from Django and which are ours.

File: src/django_components/cache_tag.py [hunk 2]
```diff
@@ -41,43 +72,76 @@
...
+    # `parent_render_id=None` makes this act as a render root, processing
+    # placeholders inline. `on_component_tree_rendered` MUST stay a no-op:
+    # a normal root passes `_render_dependencies` here, which resolves
+    # `<!-- _RENDERED -->` into `<script>`/`<link>` tags. Doing that here
+    # would break dep aggregation at the outer real root.
+    #
+    # Pop the callbacks in `finally` so a template-level loop with many
+    # {% cache %} invocations doesn't leave one set of dead lambdas per
+    # iteration in the (still-live) tree's callback dicts.
+    try:
```
Comment (JuroOravec) [inline src/django_components/cache_tag.py:111]: One code change - `component_post_render()` is wrapped in `try/finally`, and callbacks are removed after teh code is done.
